lib/msprime.c

```diff
@@ -6819,7 +6846,7 @@ msp_set_simulation_model_beta(msp_t *self, double alpha, double truncation_point
         goto out;
     }
 
-    if (truncation_point <= 0.0 || truncation_point > 1.0) {
+    if (truncation_point <= 0.0) {
```

Comment:
I think we should put a ``isfinite(truncation_point)`` check here as well. Dealing with floating point special classes is a PITA, it's much easier if we can use DBL_MAX all the way through.

Reply:
Although, I guess it's more mathematically satisfying to say ``inf`` rather than "arbitrary very big number". We'll have to sort out the annoying provenance stuff if we use ``inf``. I can put in a commit to do that if you prefer being able to use ``np.inf`` e.g. as the input value.

Reply:
I used the first thing I could make work. I'm still not very familiar with Python, and am completely unfamiliar with the C-Python interface. I'm happy to swap to `sys.foat_info.max` from above, and add the `isfinite` check.